docs/maintenance/release-management.md

```diff
@@ -146,3 +146,14 @@ only need to replace ``master`` with ``maint-x.y`` branches.
     Be aware that when the independent modules are released, they will be
     picked up immediately by new InvenioRDM installations. This should not be
     an issues since the releases MUST be backward compatible.
+
+## Trunk based Development Workflow
+
+InvenioRDM follows [Trunk-Based Development](https://trunkbaseddevelopment.com/) on the main branch (`master`) of all its modules. Changes in `master` usually won't have an out-of-the-box backwards-compatibility assurance. Things might break, but we'll fix them ASAP, since we want to be able to develop reliably.
+
+Stable releases are being maintained under their equivalent `maint` branches. For example, `v12.0.x` is maintained on the `maint-v12.0.x` branch of `invenio-app-rdm`. Those releases receive bug fixes following SemVer, e.g., `v12.0.1`, by backporting (usually via cherry-picking) the fixes from the `master` branch into the `maint` branch.
```

Comment:
```suggestion
Stable releases are being maintained under their equivalent `maint` branches. For example, `v12.0.x` is maintained on the `maint-v12.x` branch of `invenio-app-rdm`. Those releases receive bug fixes following SemVer, e.g., `v12.0.1`, by backporting (usually via cherry-picking) the fixes from the `master` branch into the `maint` branch.
```

Isn't it better to use an umbrella for all `v12.x.y` maintenance releases? That way we will not have to create a branch for each minor release

Reply:
With 3 people from 3 different orgs in agreement, I created the `maint-v12.x` branch and we'll use that one for v12 maintenance. :+1: